### PR TITLE
release: metrics-sdk metrics-api

### PR DESCRIPTION
**Pull request by xuan-cao-swi**
I'd like to suggest an alpha release for the metrics-api and metrics-sdk.

Releasing an alpha version of these metrics components could provide benefits to end-users who wish to adopt the technology and to contributors looking to gather feedback and refine the implementation based on actual usage scenarios.

### CODE REVIEW

**Comment by dmathieu** [comment]
`.alpha` is kinda unique (should there be `.alpha2`, `.alpha.2`?)
Also, the [semantic versioning spec](https://semver.org/#semantic-versioning-specification-semver) says:

> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

Based on that, it feels like releasing `0.0.1` should be good for an alpha stability.

**Comment by xuan-cao-swi** [comment]
Sample prometheus graph on histogram and counter
demo-promethesus-1
demo-promethesus-2

Here is the script that generate the graph
```ruby
require 'opentelemetry/sdk'
require 'opentelemetry-metrics-sdk'
require 'opentelemetry/exporter/otlp'
require 'vmstat'

ENV['OTEL_SERVICE_NAME'] = 'prometheus-metrics'

OpenTelemetry::SDK.configure
console_metric_exporter = OpenTelemetry::Exporter::OTLP::MetricsExporter.new
OpenTelemetry.meter_provider.add_metric_reader(console_metric_exporter)
meter = OpenTelemetry.meter_provider.meter("sample_ruby_meter_name")

mem_free_hist = meter.create_histogram("ff6fdf6a349c_mem_free", unit: 'smidgen')
mem_active_hist = meter.create_histogram("ff6fdf6a349c_mem_active", unit: 'smidgen')
mem_inactive_hist = meter.create_histogram("ff6fdf6a349c_mem_inactive", unit: 'smidgen')
mem_wired_hist = meter.create_histogram("ff6fdf6a349c_mem_wired", unit: 'smidgen')

random_number_generator_1 = meter.create_counter("ff6fdf6a349c_rand_1", unit: 'smidgen')
random_number_generator_2 = meter.create_counter("ff6fdf6a349c_rand_2", unit: 'smidgen')
random_number_generator_3 = meter.create_counter("ff6fdf6a349c_rand_3", unit: 'smidgen')
random_number_generator_4 = meter.create_counter("ff6fdf6a349c_rand_4", unit: 'smidgen')

loop do
  vmstat = Vmstat.snapshot
  memory_info = vmstat.memory

  total_memory = memory_info.wired + memory_info.active + memory_info.inactive + memory_info.free
  free = (memory_info.free.to_f / total_memory) * 100
  active = (memory_info.active.to_f / total_memory) * 100
  inactive = (memory_info.inactive.to_f / total_memory) * 100
  wired = (memory_info.wired.to_f / total_memory) * 100

  mem_free_hist.record(free.round(2), attributes: {'sample_attr' => '1'})
  mem_active_hist.record(active.round(2), attributes: {'sample_attr' => '2'})
  mem_inactive_hist.record(inactive.round(2), attributes: {'sample_attr' => '3'})
  mem_wired_hist.record(wired.round(2), attributes: {'sample_attr' => '4'})

  random_number_generator_1.add(rand(0..10), attributes: {'sample_attr' => '10'})
  random_number_generator_2.add(rand(10..20), attributes: {'sample_attr' => '11'})
  random_number_generator_3.add(rand(20..30), attributes: {'sample_attr' => '12'})
  random_number_generator_4.add(rand(30..40), attributes: {'sample_attr' => '13'})

  sleep 5
  OpenTelemetry.meter_provider.metric_readers.each(&:pull)
  sleep 10
  puts "========================= collect finished ========================="
end
```

**Comment by xuan-cao-swi** [comment]
> Based on that, it feels like releasing 0.0.1 should be good for an alpha stability.

@dmathieu I am open to change the naming for alpha release.

**Comment by hwo411** [comment]
Looking forward to having the first alpha version. We're currently integrating Opentelemetry metrics in our apps (we already have traces) and would be great to also do it for Ruby (instead of using prometheus endpoint).

**Review by mwear — COMMENTED** [review]
I agree we should make the experimental metrics sdk and api available as it's being developed. I would just like to get agreement from the other maintainers (cc @fbogsany @robertlaurin @dazuma ) on the gem name, version, and general plan for when we are ready to declare the package as stable.

I'd propose that we use the current gem names and major version of 0 as pointed out here: https://github.com/open-telemetry/opentelemetry-ruby/pull/1641#issuecomment-2141417564 to indicate the package is under development. When we decide the package is stable, we can either move the metrics api / sdk into the stable sdk and api packages, or we can leave them independent, bump their versions to 1.x.y and reference them from the stable api / sdk packages. In either case, I think we are leaving the door open if we release using the current proposal. Are there any objections or concerns anyone has with this plan?

**Review by mwear — COMMENTED** [review]
As discussed at the SIG meeting, we would like to release this later this week. Two things we would like to have before releasing are a README describing the usage and experimental state of the SDK and an issue for early adopters to provide feedback on. We plan on moving forward as discussed in https://github.com/open-telemetry/opentelemetry-ruby/pull/1641#pullrequestreview-2146114109 barring additional feedback.

**Comment by kaylareopelle** [comment]
@xuan-cao-swi  - I opened a PR with readmes for the metrics-sdk and metrics-api gems. It also includes a few small updates to the text in this PR.

https://github.com/xuan-cao-swi/opentelemetry-ruby/pull/3

File: metrics_api/lib/opentelemetry/metrics/version.rb [hunk 1]
```diff
@@ -7,6 +7,6 @@
 module OpenTelemetry
   module Metrics
     ## Current OpenTelemetry metrics version
-    VERSION = '0.0.1'
+    VERSION = '0.0.1.alpha'
```
[the remark this answers is not in the document]
Comment (kaylareopelle) [reply metrics_api/lib/opentelemetry/metrics/version.rb:10]: Will toys accept 0.0.1? IIRC the first version has to be 0.1.0.

[the remark this answers is not in the document]
Comment (mwear) [reply metrics_api/lib/opentelemetry/metrics/version.rb:10]: I updated the suggestion to be `0.1.0`. @xuan-cao-swi, PTAL.

**Review by kaylareopelle — APPROVED** [review]

**Review by mwear — APPROVED** [review]

**Comment by mwear** [comment]
I'll plan to merge this at EOD on Monday barring any new feedback.

**Pull request merged**